StormyHares/BangazonApi
# Testinglinq

## Ticket Link:
  - #11 

## Description:
  - using the URL parameter api/customers/?active=false display a list of customers that have not placed an order

## Steps to Test:

### Verify that you have all customers in the database

1. git fetch
1. git checkout  testinglinq
1. type dotnet run into terminal to start server
1. use the following url  http://localhost:5000/swagger/
1. click on the blue "GET" button listed under customer
1. without setting any options click on the "try it out" button in the bottom left. 
1. inspect the response body, you should have customers with ID 1 thought ID 20

### Add a new customer

1. click on the green "Post" under method listed under Customer - the UI should expand
1. Paste the following code into the white "Parameters" box 
    ```
     {
        "firstName": "Jimmy",
        "lastName": "Buttz",
         "lastActive": "2018-01-30T15:49:54.692Z"
      }
     ```
1. click on the "try it out" button
1. verify that you got a response code of 201 
1. inspect the response body you should receive something like the following

```
    {
            "id": 21,
            "firstName": "Jimmy",
             "lastName": "Buttz",
              "lastActive": "2018-01-30T15:49:54.692Z",
               "product": null
    }
```


### Testing ?active=false
1. click on the blue "GET" button listed under customer
1. Under parameters, find the "Active" parameter
1. in the value dropdown select "false" 
1. click on the "try it out" button in the bottom left. 
1. inspect the response body, you should have customers with ID 4,6,9, 10, 15, 18, qnd the new customer ID that we added in the previous step.
